src/protocol/packer.rs

```diff
@@ -0,0 +1,172 @@
+use std::mem::transmute;
+

```

Comment:
I don't like `packer` name. It sounds like a class name when there is no class. I believe, we should call it `pack` instead.

Reply:
So it will be used as pack::pack_int? I mean even write::write_int doesn't look good for me also. Or what was your initial idea?

Reply:
This is a common practice for Rust, so, yeah. If you don't like writing `pack::` part every time you could always `use pack`